packages/vulcan-core/lib/modules/components/App.jsx

```diff
@@ -76,12 +102,28 @@ export function parseRoute({location, onError=null}) {
   }
   
   const RouteComponent = currentRoute ? Components[currentRoute.componentName] : Components.Error404;
-  return {
+  const result = {
     currentRoute, RouteComponent, location, params,
     pathname: location.pathname,
+    url: location.pathname + location.search + location.hash,
     hash: location.hash,
     query: parseQuery(location),
   };
+  
+  if (currentRoute && currentRoute.redirect) {
```

Comment:
"currentRoute?.redirect"

Reply:
Our babel config currently is such that conditional chaining only works in the `lesswrong` package, not in the vulcan packages.